ReactWindows/ReactNative/Modules/Storage/AsyncStorageModule.cs

```diff
@@ -20,5 +52,410 @@ public override string Name
+                case JTokenType.Boolean:
+                    _dataContainer.Values[key] = token.Value<bool>();
+                    break;
+                case JTokenType.Integer:
+                    _dataContainer.Values[key] = token.Value<long>();
+                    break;
+                case JTokenType.Float:
+                    _dataContainer.Values[key] = token.Value<double>();
+                    break;
+                case JTokenType.TimeSpan:
+                    _dataContainer.Values[key] = token.Value<TimeSpan>();
+                    break;
+                case JTokenType.Guid:
+                    _dataContainer.Values[key] = token.Value<Guid>();
+                    break;
+                case JTokenType.Date:
+                    _dataContainer.Values[key] = token.ToString();
+                    addTokenTypeToContainer(key, _DataType.Date);
+                    break;
+                case JTokenType.Uri:
+                    _dataContainer.Values[key] = token.ToString();
+                    addTokenTypeToContainer(key, _DataType.Uri);
+                    break;                                
+                case JTokenType.String:
+                    _dataContainer.Values[key] = token.Value<string>();
+                    addTokenTypeToContainer(key, _DataType.String);
+                    break;
+                case JTokenType.Array:
+                    _dataContainer.Values[key] = token.ToString();
+                    addTokenTypeToContainer(key, _DataType.Array);
+                    break;
+                case JTokenType.Constructor:
+                    _dataContainer.Values[key] = token.ToString();
+                    addTokenTypeToContainer(key, _DataType.Constructor);
+                    break;
+                case JTokenType.Object:
+                    _dataContainer.Values[key] = token.ToString();
+                    addTokenTypeToContainer(key, _DataType.Object);
+                    break;              
+                case JTokenType.Raw:
+                    _dataContainer.Values[key] = token.ToString();
+                    addTokenTypeToContainer(key, _DataType.Raw);
+                    break;
+                default:
+                    Debug.Assert(false); // Not supported JToken type
+                    break;
+            }
+        }
+
+        /// <summary>
+        /// Gets related value for a specific key from <see cref="ApplicationDataContainer"/> as <see cref="JToken"/>.
+        /// </summary>
+        /// <param name="key">The key.</param>
+        private JToken getTokenFromContainer(string key)
+        {
+            object value;
+            if (_dataContainer.Values.TryGetValue(key, out value))
+            {
+                if (value is bool || value is long || value is double || value is Guid || value is TimeSpan)
+                {
+                    return JToken.FromObject(value);
+                }
+                else if (value is string)
```

Comment:
else if ((var valueStr = value as string) != null)
{
    ...

Then remove all the soft casts below.